views/cypress/tests/test.spec.js

```diff
@@ -0,0 +1,145 @@
+            });
+        cy.get('form button[type="submit"]').click();
+
+        cy.get('#tree-manage_tests')
+            .within(() => {
+                cy.contains(testDirectoryName);
+            });
+    });
+
+    it('should create test', function () {
+        cy.visit(this.urls.tests);
+
+        cy.contains(testDirectoryName).click();
+        cy.contains('Edit class');
+
+        cy.contains('New test').click();
+
+        cy.contains('Test properties').should('not.exist');
+        cy.contains('Test properties');
+
+        cy.contains('label', 'Label')
+            .parent()
+            .within(() => {
+                cy.get('input').clear().type(testName);
+            });
+        cy.get('form button[type="submit"]').click();
+
+        cy.get('#tree-manage_tests')
+            .within(() => {
+                cy.contains(testName).click();
+            });
+
+        cy.contains(testDirectoryName)
+            .parent()
+            .within(() => {
+                cy.contains(testName);
+            });
+    });
+
+    it('should update test', function() {
+        cy.get('#tree-manage_tests')
+            .within(() => {
+                cy.contains(testName).click();
+            });
+
+        cy.contains('Authoring').click();
+
+        cy.contains('h1', `${testDirectoryName} 1`)
+            .within(() => {
+                cy.get('[title="Manage test properties"]').click();
+            });
+
+        cy.get('input[name="test-identifier"]').clear().type('Test1234567890');
+        cy.get('input[name="test-title"]').clear().type('Lorem ipsum');
+
+        cy.contains('span', 'Save').click();
+        cy.contains('span', 'Manage Tests').click();
+        cy.contains('Test properties');
+        cy.contains('Test properties').should('not.exist');
+        cy.contains('Test properties');
+        cy.contains('Test Saved');
+
+        cy.contains(testName).click();
+        cy.contains('Test properties').should('not.exist');
+        cy.contains('Test properties');
+
+        cy.contains('Test Saved').should('not.exist');
+        cy.get('ul.content-action-bar')
+            .within(() => {
+                cy.contains('Authoring').click();
+            });
+
+        cy.contains('h1', 'Lorem ipsum')
+            .within(() => {
+                cy.get('[title="Manage test properties"]').click();
+            });
+        
+        cy.get('input[name="test-identifier"]').invoke('val').should('eq', 'Test1234567890');
+        cy.get('input[name="test-title"]').invoke('val').should('eq', 'Lorem ipsum');
+    });
+
+    it('should remove class', function () {
+        cy.visit(this.urls.tests);
+
+        cy.get('#tree-manage_tests')
+            .within(() => {
+                cy.contains(testDirectoryName).click();
+            });
+
+        cy.contains('Delete').click();
+        cy.get('.preview-modal-feedback.opened')
+            .within(() => {
+                cy.contains('Ok').click();
```

Comment:
Better to specify the `button` tag before `.contains`, since various elements could contain 'Ok'.